### ISSUE TITLE
Move codegen tests to run-codegen-test.yml to speed up CI verification

### ISSUE THREAD

**Issue by t0yv0** [pulumi/pulumi#9768]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

Looking at some PRs I see that we spend a lot of time in codegen tests. I'm pretty sure this is because we invoke the language toolchain and run some unit tests on top of the generated code - that is the slow part. Simply checking the code against expected output does not take that much time in Go.

Some stats from a recent PR:

```
   github.com/pulumi/pulumi/pkg/v3/codegen/nodejs	14m12.737s
   github.com/pulumi/pulumi/pkg/v3/codegen/python	5m50.882s
   github.com/pulumi/pulumi/pkg/v3/codegen/go	        5m14.161s
   github.com/pulumi/pulumi/pkg/v3/codegen/dotnet	4m48.148s
   github.com/pulumi/pulumi/pkg/v3/codegen/schema	1m42.415s
```

The suggestion is to move these tests to a more specific path-based trigger, so they would only run when codegen is modified. We already have a precedent in run-codegen-test.yml:

```
on:
  repository_dispatch:
    types: [ run-codegen-command ]
  pull_request:
    paths:
    - 'pkg/codegen/**'
    - '!pkg/codegen/docs/**'
    - '.github/workflows/run-codegen-test.yml'
```

Note the `paths`. 

This will avoid running these tests where they are useless and free up some CI runner capacity.

**Comment by AaronFriel** [comment]
@iwahbe @t0yv0 FWIW, I think we should agree that if we remove needs-triage, we need to put it in the project and/or assign it so that it's being tracked.

Perhaps we could add a project column called "priority backlog", one step between the true backlog & iteration backlog? It could contain any items we think internally we would like to work on next, say next time we're on team duty rotation and we've exhausted other work.

**Comment by iwahbe** [comment]
@AaronFriel @t0yv0 Good call. Sorry about that.

**Comment by AaronFriel** [comment]
@t0yv0 This is implemented in #10624. Codegen tests run as unit tests separate from integrations, and the pkg/codegen/nodejs tests are parallelized by assigning them to separate jobs.
